### PR TITLE
OE-6171 : Added the conditions to validate the mandatory check for th…

### PR DESCRIPTION
**Pull request by PoornimapriyaSevugarajan**
…e allergies. Also done few fixes.

### CODE REVIEW

File: protected/modules/OphCiExamination/views/default/form_Element_OphCiExamination_Allergy.php [hunk 1]
```diff
@@ -24,7 +24,10 @@
     ?>
 				<input type="hidden" name="OEModule_OphCiExamination_models_Element_OphCiExamination_Allergy[allergy_loaded]" value="1">
 				<div id="add_allergy">
-					<div class="allergies_confirm_no field-row row" <?php if ($this->patient->hasAllergyStatus() && !$this->patient->no_allergies_date) {
+					<div class="allergies_confirm_no field-row row"
+						<?php
+						$selected_allergies = Yii::app()->request->getParam('selected_allergies');
+						if ($this->patient->hasAllergyStatus() && !$this->patient->no_allergies_date && $selected_allergies) {
     echo 'style="display: none;"';
```
Comment (petergallagher) [inline protected/modules/OphCiExamination/views/default/form_Element_OphCiExamination_Allergy.php]: What is the purpose of this change? Because it seems like you want it to not show the no allergies box if someone has just posted a set of new allergies but it doesn't do that at all because the allergy status has not been set.

Comment (petergallagher) [inline protected/modules/OphCiExamination/views/default/form_Element_OphCiExamination_Allergy.php]: Also, can you tidy up the file as well, as there are closing php tags in very strange places and terrible indentation.

Comment (PoornimapriyaSevugarajan) [inline protected/modules/OphCiExamination/views/default/form_Element_OphCiExamination_Allergy.php]: This is necessitated to fix the following implications, when there are some allergies associated to the patient already, and we have removed all the allergies. Also, no allergies check box is empty, but in the backend, we will be having allergies that removed now, will be present in the event. In this case, its not showing No allergies checkbox. So, I have approached in the reverse way, to fix the bug. I will tidy up the file.
